lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -282,6 +282,12 @@ namespace Js
         }
     }
 
+    ArrayBuffer * ArrayBuffer::GetAsArrayBuffer()
+    {
+        AssertOrFailFast(VarIs<ArrayBuffer>(this));
```

Comment:
nit: This VarIs should not be possible to fail without already having significantly violated the type system.

Reply:
Rather than removing existing checks, let's change them to a version of the VarIs call that necessarily failfasts if the check doesn't pass. If we don't have information that this is affecting perf, then we might as well keep the checks in.